mos_tests/nova/nova_tests.py

```diff
@@ -466,3 +484,106 @@ def test_2238776_NetworkConnectivityToVMDuringLiveMigration(self):
+        self.keys.append(keys)
+        private_key = paramiko.RSAKey.from_private_key(six.StringIO(str(
+            keys.private_key)))
+        inst = common_functions.create_instance(self.nova,
+                                                "inst_2238776_{}"
+                                                .format(flavor.name),
+                                                flavor.id, net,
+                                                [self.sec_group.name],
+                                                image_id=image_id,
+                                                key_name='key_2238776',
+                                                inst_list=self.instances)
+        inst.add_floating_ip(floating_ip.ip)
+        ping = common_functions.ping_command(floating_ip.ip, i=10)
+        self.assertTrue(ping, "Instance is not reachable")
+        out = []
+        with SSHClient(host=floating_ip.ip, username="cirros", password=None,
+                       private_keys=[private_key]) as vm_r:
+            out.append(vm_r.execute("sudo sh -c 'date > /timestamp.txt'"))
+            out.append(vm_r.execute("sudo sh -c 'date > /mnt/timestamp.txt'"))
+            out.append(vm_r.execute("sudo -i cat /timestamp.txt"))
+            out.append(vm_r.execute("sudo -i cat /mnt/timestamp.txt"))
+
+        for i in out:
+            if i['stderr'] != []:
+                raise Exception("ssh commands were executed with errors")
+
+        root_data = out[-2]['stdout'][0]
+        ephem_data = out[-1]['stdout'][0]
+
+        # live migration
+        hypervisors = {h.hypervisor_hostname: h for h in
+                       self.nova.hypervisors.list()}
+        old_hyper = getattr(inst, "OS-EXT-SRV-ATTR:hypervisor_hostname")
+        new_hyper = [h for h in hypervisors.keys() if h != old_hyper][0]
+        ping = subprocess.Popen(["/bin/ping", "-c100", "-i1", floating_ip.ip],
+                                stdout=subprocess.PIPE)
+        self.nova.servers.live_migrate(inst, new_hyper, block_migration=False,
+                                       disk_over_commit=False)
+        inst = self.nova.servers.get(inst.id)
+        timeout = 10
+        end_time = time() + 60 * timeout
+        while getattr(inst, "OS-EXT-SRV-ATTR:hypervisor_hostname") != \
+                new_hyper:
+            if time() > end_time:
+                raise AssertionError(
+                    "Hypervisor is not changed after live migration")
+            sleep(1)
+            inst = self.nova.servers.get(inst.id)
+        self.assertEqual(inst.status, 'ACTIVE')
+        ping.wait()
+        output = ping.stdout.read().split('\n')[-3].split()
+        packets = {'transmitted': int(output[0]), 'received': int(output[3])}
+        loss = packets['transmitted'] - packets['received']
+        if loss > 5:
+            raise AssertionError(
+                "Packets loss exceeds the limit, {} packets were lost".format(
```

Comment:
msg = "Packets loss exceeds the limit, {} packets were lost"
raise AssertionError(msg.format(loss))